drivers/serial/uart_ns16550.c

```diff
@@ -246,13 +226,14 @@ struct uart_ns16550_device_config {
 #if defined(CONFIG_PINCTRL)
 	const struct pinctrl_dev_config *pincfg;
 #endif
+#if defined(CONFIG_UART_NS16550_ACCESS_IOPORT) || defined(CONFIG_UART_NS16550_SIMULT_ACCESS)
```

Comment:
If `CONFIG_UART_NS16550_ACCESS_IOPORT` is set, then doesn't that mean that every instance is assumed to be using IO port access? In that case this variable is then also unnecessary, i.e. it should only depend on `SIMULT_ACCESS`, right?